binding/react/src/use_cheetah.ts

```diff
@@ -0,0 +1,176 @@
+/*
+  Copyright 2023 Picovoice Inc.
+
+  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+  file accompanying this source.
+
+  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+  specific language governing permissions and limitations under the License.
+*/
+
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+import { WebVoiceProcessor } from '@picovoice/web-voice-processor';
+
+import {
+  CheetahModel,
+  CheetahOptions,
+  CheetahTranscript,
+  CheetahWorker,
+} from '@picovoice/cheetah-web';
+
+export const useCheetah = (): {
+  result: {
+    partialTranscript: string;
```

Comment:
I think we may just want to call this `transcript`. The `partialTranscript` naming is better for when you are distinguishing a partial from a final, but in this case we have `isComplete` which tells us when the transcript is complete. Otherwise it is implied that it is a partial transcript.